mkdocs.yml

```diff
@@ -0,0 +1,100 @@
+site_name: Temporian
+repo_url: https://github.com/google/temporian
+
+# Theme
+theme:
+  name: "material"
+  icon:
+    repo: fontawesome/brands/github
+  features:
+    - navigation.tabs
+    - navigation.tracking
+    - navigation.sections
+    - navigation.instant
+    - navigation.indexes
+  palette:
+    # Palette toggle for dark mode
+    - scheme: slate
+      toggle:
+        icon: material/brightness-4
+        name: Switch to light mode
+      primary: black
+      accent: white
+    # Palette toggle for light mode
+    - scheme: default
+      toggle:
+        icon: material/brightness-7
+        name: Switch to dark mode
+      primary: black
+      accent: white
+
+# Extra CSS
+extra_css:
+  # TODO: continue working on this css customization for temporian
+  - css/custom.css
+
+# Navigation bar
+nav:
+  - Home: index.md
+  - Getting Started: getting_started.md
+  - Tutorials: tutorials/
+  - Code Reference: reference/ # generated by gen-files + literate-nav
+
+# Plugins
+plugins:
+  - search
+  - autorefs
+  - gen-files:
+      scripts:
+        # Generate the index page from the README.md.
+        - docs/gen_index.py
+        # Generate the reference pages from the docstrings.
+        - docs/gen_ref_pages.py
+  - literate-nav:
+      nav_file: SUMMARY.md
+    # Configuration to generate the reference pages from the docstrings.
+  - mkdocstrings:
```

Comment:
Great work! Is it viable to only show the docstrings/reference of the public functions/classes with which the users will interact? Currently, we have references for even the tests. I know there are ways to specify which functions/classes you want to show on the code. It may be too tedious to do this way but I don't know what alternative we have.

For more information we can check on this page: https://mkdocstrings.github.io/usage/

Reply:
Another thing to keep in mind is if these types of documentation are supposed to document all the code not only the public API functions/classes. If that is the case, you can ignore my comment.

Reply:
We could consider having separate sections: users and contributors. So users will not be exposed to the whole code base when searching for documentation. And still have the documentation necessary if anyone wants to understand better the library and contribute to it.

Reply:
https://github.com/google/temporian/issues/104 opened this issue an hour ago stating we should remove test modules from docs 👍🏼 and agree that splitting public/internal ones could be a good idea too!